### PR TITLE
feat(pocket): adding permissions to Pockets Prefect flow to access new tab datasets

### PR DESCRIPTION
**Pull request by chelseybeck**
Prefect needs dataset access in order to run flows extracting data from glean new tab datasets

[MSDO-65](https://mozilla-hub.atlassian.net/browse/MSDO-65)

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2178)

[MSDO-65]: https://mozilla-hub.atlassian.net/browse/MSDO-65?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/firefox_desktop/dataset_metadata.yaml [hunk 1]
```diff
@@ -8,3 +8,5 @@ workgroup_access:
 - role: roles/bigquery.dataViewer
   members:
   - workgroup:mozilla-confidential
+  - workgroup:pocket/prefect
+#adding comment to add a change
```
Comment (chelseybeck) [inline sql/moz-fx-data-shared-prod/firefox_desktop/dataset_metadata.yaml:12]:
```suggestion
```

File: sql/moz-fx-data-shared-prod/firefox_desktop/dataset_metadata.yaml [hunk 2]
```diff
@@ -8,3 +8,4 @@ workgroup_access:
 - role: roles/bigquery.dataViewer
   members:
   - workgroup:mozilla-confidential
+  - workgroup:pocket/prefect
```
Comment (chelseybeck) [inline sql/moz-fx-data-shared-prod/firefox_desktop/dataset_metadata.yaml:11]:
```suggestion
  - workgroup:pocket/prefect
  
```

**Comment by whd** [comment]
This dataset (or rather, the underlying ingestion datasets) fall under the [ingestion dataset with table-level access by doctype](https://mozilla-hub.atlassian.net/wiki/spaces/SRE/pages/27924789/Data+Access+Workgroups#DataAccessWorkgroups-Exampleingestiondatasetwithtable-levelaccessbydoctype(restrictingaccesstocertaindoctypeswithinanotherwiseworkgroup%3Amozilla-confidentialdataset)) pattern. This means a PR like this has two problems:

1. unless each view in the views dataset is authorized, access granted to the user-facing dataset is insufficient (you will have access to query views but hit permissions errors when querying them)
3. because there are workgroup-confidential tables in these datasets (specifically [these pings](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-shared/tf/prod/envs/prod/bigquery-new/namespaces.auto.tfvars.json#L415-L430)), you would need to get approval from an owner or manager of [workgroup:contextual-services](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-shared/tf/global/access-groups/workgroups.tf#L591) in order to grant dataset-level access.

If you're specifically interested in `newtab` pings, I suggest [annotating only those pings](https://mozilla-hub.atlassian.net/wiki/spaces/SRE/pages/27924789/Data+Access+Workgroups#DataAccessWorkgroups-Workgroupaccessforautomationandapplicationserviceaccounts) with the relevant `workgroup_access`. That would mean copying some of the auto-generated metadata in https://github.com/mozilla/private-bigquery-etl/blob/private-generated-sql/sql/moz-fx-data-shared-prod/firefox_desktop/newtab/metadata.yaml to the `main` branch here and updating its `workgroup_access` to include `workgroup:pocket/prefect` while also filing a PR to update https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-shared/tf/prod/envs/prod/bigquery-new/namespaces.auto.tfvars.json#L414 to include `newtab` with the specific permissions.

**Comment by chelseybeck** [comment]
closing this pr in favor of the above pattern

**Pull request closed without merging**